bear.py

```diff
@@ -0,0 +1,176 @@
+            else:
+                # No one defeats - fight
+                my_fighting_coef = generate_fighting_coef()
+                his_fighting_coef = generate_fighting_coef()
+
+                while my_fighting_coef == his_fighting_coef:
+                    my_fighting_coef = generate_fighting_coef()
+                    his_fighting_coef = generate_fighting_coef()
+
+                # I win
+                if my_fighting_coef > his_fighting_coef:
+                    end_of_fight(field, self, another_bear, my_fighting_coef, his_fighting_coef)
+                    print >> out_file, 'B{} wins '.format(self.name)
+                # I loose
+                if his_fighting_coef > my_fighting_coef:
+                    end_of_fight(field, another_bear, self, his_fighting_coef, my_fighting_coef)
+                    print >> out_file, 'B{} looses '.format(self.name)
+        else:
+            # Aggression level doesn't let fighting
+            print >> out_file, "Agression not enough for fighting, go to random cell"
+            self.go_to_random_cell(field)
+
+        field.draw_field()
+        field.draw_bears_data()
+
+    def move(self, field):
+
+        new_row = 0
+        new_col = 0
+        mov_log = ""
+        do_move = random.random()
+
+        if do_move > self.activity_level:
+            print >> out_file, 'B{} rests'.format(self.name)
+            self.last_action == constants.REST
+            return
+
+        if self.last_action == constants.REST:
+            self.last_direction = random.randint(0, 3)
+
+        self.last_action = constants.MOVE
+
+        if self.last_direction == constants.MOVEMENT_UP:
+            mov_log = mov_log + " UP "
+            if self.cell_row == 0:
+                new_row = constants.MAX_ROW - 1
+            else:
+                new_row = self.cell_row - 1
+            new_col = self.cell_col
+
+        if self.last_direction == constants.MOVEMENT_DOWN:
+            mov_log = mov_log + " DOWN "
+            if self.cell_row == constants.MAX_ROW - 1:
+                new_row = 0
+            else:
+                new_row = self.cell_row + 1
+            new_col = self.cell_col
+
+        if self.last_direction == constants.MOVEMENT_LEFT:
+            mov_log = mov_log + " LEFT "
+            if self.cell_col == 0:
+                new_col = constants.MAX_COL - 1
+            else:
+                new_col = self.cell_col - 1
+            new_row = self.cell_row
+
+        if self.last_direction == constants.MOVEMENT_RIGHT:
+            mov_log = mov_log + " RIGHT "
+            if self.cell_col == constants.MAX_COL - 1:
+                new_col = 0
+            else:
+                new_col = self.cell_col + 1
+            new_row = self.cell_row
+
+        print >> out_file, 'B{} moves {} from r{} c{} to r{} c{}'.format(self.name, mov_log, self.cell_row, self.cell_col, new_row, new_col)
```

Comment:
I'm not a big fan of all these prints within the class
Try reading about the module [logging](https://docs.python.org/2.7/library/logging.html)

Reply:
ok, I will, I needed it for debug and visualization